### ISSUE TITLE
Bug: [no-duplicate-enum-values] Should check static template literal

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#10674]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=KYOwrgtgBAolDeAoKUCCUC8UBErsBpkoAhTKAA1XMJQGEzLzEBfRIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNZjGyAQ3yJ6TWAFtaAN0HxYKdFETRoHaJHBgAviG1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
enum E {
  A = "A",
  B = `A`, // no error
  C = `A` // no error
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-duplicate-enum-values": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

```ts
enum E {
  A = "A",
  B = `A`, // Error: Duplicate enum member value `A`
  C = `A` // Error: Duplicate enum member value `A`
}
```

### Actual Result

```ts
enum E {
  A = "A",
  B = `A`, // no error
  C = `A` // no error
}
```

### Additional Info
-  #10675
_No response_
